test/core/xds/certificate_provider_store_test.cc

```diff
@@ -0,0 +1,129 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#include <grpc/support/port_platform.h>
+
+#include <gmock/gmock.h>
+
+#include "src/core/ext/xds/certificate_provider_registry.h"
+#include "src/core/ext/xds/certificate_provider_store.h"
+
+#include "test/core/util/test_config.h"
+
+namespace grpc_core {
+namespace testing {
+namespace {
+
+class FakeCertificateProvider : public grpc_tls_certificate_provider {
+ public:
+  RefCountedPtr<grpc_tls_certificate_distributor> distributor() const {
+    // never called
+    GPR_ASSERT(0);
+    return nullptr;
+  }
+};
+
+class FakeCertificateProviderFactory1 : public CertificateProviderFactory {
+ public:
+  class Config : public CertificateProviderFactory::Config {
+   public:
+    const char* name() const override { return "fake1"; }
+  };
+
+  const char* name() const override { return "fake1"; }
+
+  RefCountedPtr<CertificateProviderFactory::Config>
+  CreateCertificateProviderConfig(const Json& config_json,
+                                  grpc_error** error) override {
+    return MakeRefCounted<Config>();
+  }
+
+  RefCountedPtr<grpc_tls_certificate_provider> CreateCertificateProvider(
+      RefCountedPtr<CertificateProviderFactory::Config> config) override {
+    return MakeRefCounted<FakeCertificateProvider>();
+  }
+};
+
+class FakeCertificateProviderFactory2 : public CertificateProviderFactory {
+ public:
+  class Config : public CertificateProviderFactory::Config {
+   public:
+    const char* name() const override { return "fake2"; }
+  };
+
+  const char* name() const override { return "fake2"; }
+
+  RefCountedPtr<CertificateProviderFactory::Config>
+  CreateCertificateProviderConfig(const Json& config_json,
+                                  grpc_error** error) override {
+    return MakeRefCounted<Config>();
+  }
+
+  RefCountedPtr<grpc_tls_certificate_provider> CreateCertificateProvider(
+      RefCountedPtr<CertificateProviderFactory::Config> config) override {
+    return MakeRefCounted<FakeCertificateProvider>();
+  }
+};
+
+TEST(CertificateProviderStoreTest, Basic) {
+  // Set up factories. (Register only one of the factories.)
+  auto* fake_factory_1 = new FakeCertificateProviderFactory1;
+  CertificateProviderRegistry::RegisterCertificateProviderFactory(
+      std::unique_ptr<CertificateProviderFactory>(fake_factory_1));
+  auto fake_factory_2 = absl::make_unique<FakeCertificateProviderFactory2>();
+  // Set up store
+  CertificateProviderStore::PluginDefinitionMap map;
```

Comment:
This can be a single literal:

```
CertificateProviderStore::PluginDefinitionMap map = {
  {"fake_plugin_1",
   {"fake1", fake_factory_1->CreateCertificateProviderConfig(
                            Json::Object(), nullptr)}},
  {"fake_plugin_2",
   {"fake2", fake_factory_2->CreateCertificateProviderConfig(
                            Json::Object(), nullptr)}},
  {"fake_plugin_3",
   {"fake1", fake_factory_1->CreateCertificateProviderConfig(
                            Json::Object(), nullptr)}},
};
```

Reply:
Done.